test/client_ipc.cpp

```diff
@@ -6,34 +6,36 @@
 * The full license is in the file LICENSE, distributed with this software. *
 ****************************************************************************/
 
-// TODO: this client should be reimplemented with
-// the new client framework. Notice that it was not
-// tested in the CI but was here for trying to reproduce
-// some issue with IPC.
-
-/*#include <iostream>
+#include <iostream>
 
 #include "zmq_addon.hpp"
 #include "nlohmann/json.hpp"
 #include "xeus/xguid.hpp"
 #include "xeus/xmessage.hpp"
-#include "xeus-zmq/xauthentication.hpp"
-#include "xeus-zmq/xmiddleware.hpp"
-#include "xeus-zmq/xzmq_serializer.hpp"
+#include "xeus/xkernel_configuration.hpp"
+#include "xeus-zmq/xzmq_context.hpp"
+#include "xipc_client.hpp"
 
 namespace nl = nlohmann;
 
 int main(int, char**)
 {
-    zmq::context_t context;
-    zmq::socket_t cli(context, zmq::socket_type::dealer);
-    xeus::xguid socket_id = xeus::new_xguid();
-    //cli.set(ZMQ_IDENTITY, socket_id.c_str(), socket_id.size());
-    cli.set(zmq::sockopt::linger, xeus::get_socket_linger());
-    cli.connect(xeus::get_end_point("ipc", "localhost", "shell"));
+    auto context_ptr = xeus::make_zmq_context();
+    xeus::xcontext& context = *context_ptr;
+    xeus::xconfiguration config;
+    config.m_transport = "ipc";
+    config.m_ip = "localhost";
+    config.m_control_port = "control";
+    config.m_shell_port = "shell";
+    config.m_stdin_port = "stdin";
+    config.m_iopub_port = "iopub";
+    config.m_hb_port = "heartbeat";
+    config.m_signature_scheme = "none";
+    config.m_key = "";
 
-    auto auth = xeus::make_xauthentication("none", "");
+    xeus::xipc_client ipc_client(context, config);
```

Comment:
Here you could write `xeus::xipc_client ipc_client(*context_ptr, config)`;

Reply:
Addressed